src/main/jbake/content/developer/jms/advanced.adoc

```diff
@@ -0,0 +1,109 @@
+= JMS Advanced
+:jbake-date: 2016-03-17
+:jbake-type: page
+:jbake-status: published
+:jbake-tomeepdf:
+
+
+=== Configuring the Container tomee.xml
+
+===== Queue
+
+[source,xml]
+----
+<Resource id="Foo" type="javax.jms.Queue">
+    destination = Specifies the name of the queue
+</Resource>
+----
+
+===== Topic
+
+[source,xml]
+----
+<Resource id="Foo" type="javax.jms.Topic">
+    destination = Specifies the name of the topic
+</Resource>
+----
+
+
+===== ConnectionFactory
+
+[source,xml]
+----
+<Resource id="Foo" type="javax.jms.ConnectionFactory">
+    ResourceAdapter = Default value is Default JMS Resource Adapter.
+    TransactionSupport = Specifies if the connection is enrolled in global transaction, allowed values: xa, local or none. Default value is xa.
+    PoolMaxSize = Maximum number of physical connection to the ActiveMQ broker. Default value is 10.
+    PoolMinSize  = Minimum number of physical connection to the ActiveMQ broker. Default value is 0.
+    ConnectionMaxWaitMilliseconds = Maximum amount of time to wait for a connection. Default value is 5000.
+    ConnectionMaxIdleMinutes = Maximum amount of time a connection can be idle before being reclaimed. Default value is 15.
+</Resource>
+----
+
+===== MESSAGE
+
+[source,xml]
+----
+<Container id="Foo" type="MESSAGE">
+    ResourceAdapter = The resource adapter delivers messages to the container. Default value is Default JMS Resource Adapter.
+    MessageListenerInterface = Specifies the message listener interface handled by this container. Default value is javax.jms.MessageListener.
+    ActivationSpecClass = Specifies the activation spec class. Default value is org.apache.activemq.ra.ActiveMQActivationSpec.
+    InstanceLimit = Specifies the maximum number of bean instances that are allowed to exist for each MDB deployment. Default value is 10.
+</Container>
+----
+
+
+=== Activation Message
+
+A JMS activation specification is associated with one or more message-driven beans and provides the configuration necessary for them to receive messages.
```

Comment:
Also note that these can be specified by:
* annotations on the bean (example would be great)
* in ejb-jar.xml
* as properties (with the prefix of "activation.") on bean in openejb-jar.xml
* properties (with the prefix of "activation.") on a MDB container in `tomee.xml`
* as system properties

In terms of priority:

* bean specific activation properties override global properties
* annotations are overridden by ejb-jar.xml values, which in turn are overridden by system properties.